### PR TITLE
Magpietts 2508 Attention mask bug fix

### PR DESCRIPTION
**Pull request by paarthneekhara**
Added a test as well. Handling masking in batched inference for attention and convolutional layer.

### CODE REVIEW

**Review by blisc — APPROVED** [review]

**Comment by rfejgin** [comment]
Looks good to me. 

It would be good to also add a similar test for cross attention. Doesn't have to be in this PR.

File: tests/collections/tts/modules/test_transformer_2501.py [hunk 1]
```diff
@@ -795,3 +800,58 @@ def test_forward_causal_self_attn_and_has_xattn(self):
...
+        query_tensor2_padded = torch.cat([query_tensor2, padding_tensor], dim=1)
+        lengths = torch.tensor([self.long_length, self.short_length])
+        mask_batched = get_mask_from_lengths(lengths)
+
+        query_batched = torch.cat([query_tensor1, query_tensor2_padded], dim=0)
+
+        mask_bs1_1 = torch.ones(1, self.long_length)
+        mask_bs1_2 = torch.ones(1, self.short_length)
+
+        for is_causal in [True, False]:
```
Comment (rfejgin) [inline tests/collections/tts/modules/test_transformer_2501.py:834]: Nice coverage of the various config combinations!
